compiler/quilt/tools/command.py

```diff
@@ -1326,17 +1332,85 @@ def iter_filename_map(node):
         # Technically successful, but with nothing to do.
         # package may have no file nodes, or user may have filtered out all applicable targets.
         # -- should we consider it an error and raise?
-        print("No files to export.")
+        if not silent:
+            print("No files to export.")
         return
 
+    # ensure output path is writable.  I'd just check stat, but this is fully portable.
+    try:
+        output_path.mkdir(exist_ok=True)  # could be '.'
+        with tempfile.TemporaryFile(dir=str(output_path), prefix="quilt-export-write-test-", suffix='.tmp'):
```

Comment:
Add a TODO: performance hit due to write amplification

Reply:
Will do, but that's a one-time hit, per-export, not in a loop or anything.